### PR TITLE
Add question 5.1 - What kind of space?

### PR DESCRIPTION
**Pull request by injms**
This has been mainly ~copied~ updated from 'What kind of medical equipment?' question, with the content updated, since the questions are similar in terms of the form fields and logic.

The logic for question 5 ('Can you offer space?') have been updated to point to this question if 'yes', and go to the next question if 'No'.

### CODE REVIEW

**Review by bilbof — COMMENTED** [review]
Looks good! Please can we have a test for this one?

File: app/views/coronavirus_form/offer_space_type.erb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<% content_for :meta_tags do %>
+  <meta name="description" content="<%= t('coronavirus_form.what_kind_of_space.title') %>" />
+<% end %>
+
+<% content_for :back_link do %>
+  <%= render "govuk_publishing_components/components/back_link", { href: previous_path } %>
+<% end %>
+
+<%= form_tag({},
+  "data-module": "track-coronavirus-form-business-offer-space",
```
Comment (richardTowers) [inline app/views/coronavirus_form/offer_space_type.erb]: Should this be different from the `offer_space` view?

File: app/controllers/coronavirus_form/offer_space_controller.rb [hunk 2]
```diff
@@ -22,6 +22,8 @@ def submit
       render "coronavirus_form/#{PAGE}"
     elsif session["check_answers_seen"]
       redirect_to controller: "coronavirus_form/check_answers", action: "show"
+    elsif offer_space === "Yes"
```
Comment (huwd) [inline app/controllers/coronavirus_form/offer_space_controller.rb]: I'm a little unsure if `===` or `==` is better pratice here.
I always get confused coming back from JS.

I know that `===` is implimented differently depending on the parent class.
But this seems to work here with the string matching, so feel free to ignore if this is pedentry!

File: spec/controllers/coronavirus_form/offer_space_controller_spec.rb [hunk 3]
```diff
@@ -24,11 +24,16 @@
       expect(session[session_key]).to eq selected
     end
 
-    it "redirects to next step for a permitted response" do
-      post :submit, params: { offer_space: selected }
+    it "redirects to next step for a 'No' response" do
```
Comment (huwd) [inline spec/controllers/coronavirus_form/offer_space_controller_spec.rb:27]: Much clearer, will update my tests to use that!

**Review by huwd — APPROVED** [review]

File: app/views/coronavirus_form/offer_space_type.erb (hunk 1 above)
Comment (huwd) [reply app/views/coronavirus_form/offer_space_type.erb]: I think so, this gets rendered as a `data-attribute` in the HTML,
So we're using html attribute syntax wich is all kebabs i believe

Comment (injms) [reply app/views/coronavirus_form/offer_space_type.erb]: Yes, yes it should. Thank you!

File: app/controllers/coronavirus_form/offer_space_controller.rb (hunk 2 above)
Comment (injms) [reply app/controllers/coronavirus_form/offer_space_controller.rb]: Would `.eql?` be more Ruby?

Comment (injms) [reply app/controllers/coronavirus_form/offer_space_controller.rb]: I've changed it to `.eql?` :)

File: spec/controllers/coronavirus_form/offer_space_type_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    it "validates a valid option is chosen" do
+      post :submit, params: { offer_space: "<script></script>" }
+
+      expect(response).to render_template(current_template)
+    end
+
+    it "redirects to check your answers if check your answers previously seen" do
+      session[:check_answers_seen] = true
+      post :submit, params: { offer_space: selected }
```
Comment (bilbof) [inline spec/controllers/coronavirus_form/offer_space_type_controller_spec.rb]: This should be `offer_space_type`

Comment (injms) [reply spec/controllers/coronavirus_form/offer_space_type_controller_spec.rb]: Yes!

**Review by huwd — APPROVED** [review]

**Comment by bilbof** [comment]
🚢 :shipit:

**Pull request merged**